### PR TITLE
Read-only root filesystem and tmpfs size support in cli

### PR DESCRIPTION
**Pull request by skinowski**
TODO: TmpFs not yet in fn_go. Will need to add/update fn_go and add TmpFsSize to routes.go.

### CODE REVIEW

**Comment by skinowski** [comment]
fn recently switched to read-only root filesystem by default. To follow up, CLI needs to make read-only root default as well. This PR introduces read-only root filesystem and an option to opt out in 'fn run' command.

File: test.sh [hunk 1]
```diff
@@ -39,21 +39,26 @@ CONTAINER_ID=$($fn start -d)
...
 
-	if [ ${RESULT} -ne 0 ]; then
-		sleep 1
-		TRIES=$((${TRIES}-1))
+	if [ ${RESULT} -eq 0 ]; then
+		break
+	else
+		echo "Going to try ${FN_API_URL} one more time..."
 	fi
-	break
```
Comment (skinowski) [inline test.sh]: Bug fix here...

File: test.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ export fn="${CUR_DIR}/fn"
 
 
 #on CI these can take a while
-go test -v $(go list ./... |  grep -pv "^github.com/fnproject/cli/test$")
```
Comment (skinowski) [inline test.sh]: Linux does not like '-p' and sure if follow-sym links is needed here. On mac-os-x, man page of grep says:

"-p      If -R is specified, no symbolic links are followed.  This is the default."

which seems unnecessary in test.sh above.

File: run.go [hunk 3]
```diff
@@ -190,6 +194,16 @@ func runff(ff *funcfile, stdin io.Reader, stdout, stderr io.Writer, method strin
 		}
 	}
 
+	if !rwRoot {
+		sh = append(sh, "--read-only")
+	}
+
+	if ff.TmpFsSize != 0 {
```
Comment (skinowski) [inline run.go]: Let's always try to add a rw tmpfs here. This is also the default behavior in fn agent now.

**Comment by rdallman** [comment]
fn test/run are gone now, gonna close

**Pull request closed without merging**